File: javascript/strain/strain.js
var keep = function(collection, callback) {
    return collection.filter(function(item) {
        return callback(item)
    })
};

var discard = function(collection, callback) {
    return collection.filter(function(item) {
        return !callback(item)
    })
};

module.exports = { keep, discard };

File: ruby/strain/strain.rb
module Enumerable
  def keep
    each_with_object([]) do |item, keepers|
      keepers << item if yield(item)
    end
  end

  def discard
    each_with_object([]) do |item, keepers|
      keepers << item unless yield(item)
    end
  end
end
